src/ax_interface/mib.py

```diff
@@ -37,7 +41,18 @@ def __init__(self):
             else:
                 self.update_counter += 1
             # run the background update task
-            self.update_data()
+            try:
+                self.update_data()
+            except (KeyboardInterrupt, SystemExit):
+                # They are exception thrown when Ctrl+C or sys.exit() happens
+                raise
+            except:
+                # Any other exception or error, log it and keep running
+                ex = sys.exc_info()[0]
+                exstr = "MIBUpdater.start() caught an unexpected exception: {}".format(str(ex))
+                print(exstr, flush=True)
+                logger.exception(exstr)
```

Comment:
Already fixed in the 2nd commit. Could you double check?

---
In reply to: [121173665](https://github.com/Azure/sonic-snmpagent/pull/32#discussion_r121173665) [](ancestors = 121173665)